Author: Sergei Golovan
Description: Patch disables loading erlang-edoc.el and erldoc.el for XEmacs
 because they require xmltok and cl-lib which aren't there.
Last-Modified: Tue, 24 Apr 2018 13:58:13 +0300
Debian-Bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=852823

--- a/lib/tools/emacs/erlang-start.el
+++ b/lib/tools/emacs/erlang-start.el
@@ -78,21 +78,24 @@
 (autoload 'erlang-find-tag-other-window "erlang"
   "Like `find-tag-other-window'.  Capable of retreiving Erlang modules.")
 
+
 ;;
 ;; Declare functions in "erlang-edoc.el".
 ;;
 
-(autoload 'erlang-edoc-mode "erlang-edoc" "Toggle Erlang-Edoc mode on or off." t)
+(defconst erlang-xemacs-p (string-match "Lucid\\|XEmacs" emacs-version)
+  "Non-nil when running under XEmacs or Lucid Emacs.")
+(if (not erlang-xemacs-p) (autoload 'erlang-edoc-mode "erlang-edoc" "Toggle Erlang-Edoc mode on or off." t))
 
 ;;
 ;; Declare functions in "erldoc.el".
 ;;
 
-(autoload 'erldoc-browse "erldoc" "\n\n(fn MFA)" t nil)
-(autoload 'erldoc-browse-topic "erldoc" "\n\n(fn TOPIC)" t nil)
-(autoload 'erldoc-apropos "erldoc" "\n\n(fn PATTERN)" t nil)
-(autoload 'erldoc-eldoc-function "erldoc" "\
-A function suitable for `eldoc-documentation-function'.\n\n(fn)" nil nil)
+(if (not erlang-xemacs-p) (autoload 'erldoc-browse "erldoc" "\n\n(fn MFA)" t nil))
+(if (not erlang-xemacs-p) (autoload 'erldoc-browse-topic "erldoc" "\n\n(fn TOPIC)" t nil))
+(if (not erlang-xemacs-p) (autoload 'erldoc-apropos "erldoc" "\n\n(fn PATTERN)" t nil))
+(if (not erlang-xemacs-p) (autoload 'erldoc-eldoc-function "erldoc" "\
+A function suitable for `eldoc-documentation-function'.\n\n(fn)" nil nil))
 
 ;;
 ;; Associate files extensions ".erl" and ".hrl" with Erlang mode.
